src/components/GlobalPreferences/Notifications/NotificationsVerify.js

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import { ButtonText, LoadingRing } from '@aragon/ui'
+import { verifyEmailToken } from './notification-service-api'
+import { VERIFY_SUBSECTION, ExpiredTokenError } from './constants'
+import NotificationsInfoBox, {
+  ICON_SUCCESS,
+  ICON_ERROR,
+  ICON_NEUTRAL,
+} from './NotificationsInfoBox'
+
+export function NotificationsVerify({
+  subsection,
+  onTokenChange,
+  onEmailChange,
+  navigateToNotifications,
+}) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [verified, setVerified] = useState(false)
+  const [error, setError] = useState(null)
+
+  const handleResetAccount = useCallback(() => {
+    // In error states we can clear email and token to reset the state
+    onEmailChange(null)
+    onTokenChange(null)
+    navigateToNotifications()
+  }, [onEmailChange, onTokenChange, navigateToNotifications])
+
+  useEffect(() => {
+    let cancelled = false
+    // Parse token from subsection /verify/[TOKEN] -> [TOKEN]
+    const token = subsection.substring(VERIFY_SUBSECTION.length)
+    verifyEmailToken(token)
+      .then(longLivedToken => {
+        if (!cancelled) {
+          setVerified(true)
+          setIsFetching(false)
+          onTokenChange(longLivedToken)
+        }
+        return longLivedToken
+      })
+      .catch(e => {
+        if (!cancelled) {
+          setError(e)
+          setIsFetching(false)
+          setVerified(false)
+        }
+      })
+    return () => {
+      cancelled = true
+    }
+  }, [subsection, onTokenChange, onEmailChange])
+
+  if (isFetching) {
+    return (
+      <NotificationsInfoBox header="Verifying">
+        <div
+          css={`
+            display: flex;
+            justify-content: center;
+            align-items: center;
+          `}
+        >
+          <LoadingRing />
+        </div>
+      </NotificationsInfoBox>
+    )
+  }
+  if (verified) {
```

Comment:
Mot for this iteration but maybe we could redirect the user after N seconds of this screen to Notification preferences?

Reply:
Added in 0f788d09c63480c12fd5147a444b87ace699f3d5 after 10 seconds